File: PF2-web5/web5-vue/web2-vue/src/types/Requirement.ts
export class Requirement {
  requirement_no: number;
  requirement_context: string;
  requirement_x: number;
  requirement_y: number;
  requirement_h: number;
  requirement_w: number;

  static newRequirement(no: number, name: string, x: number, y: number, h: number, w: number) {
    let req = new Requirement();
    req.requirement_no = no;
    req.requirement_context = name;
    req.requirement_x = x;
    req.requirement_y = y;
    req.requirement_h = h;
    req.requirement_w = w;
    return req;
  }

  static newRequirementWithOld(old: Requirement, name: string) {
    let req = new Requirement();
    req.requirement_no = old.requirement_no;
    req.requirement_context = name;
    req.requirement_x = old.requirement_x;
    req.requirement_y = old.requirement_y;
    req.requirement_h = old.requirement_h;
    req.requirement_w = old.requirement_w;
    return req;
  }

  static copyRequirement(old: Requirement) {
    let req = new Requirement();
    req.requirement_no = old.requirement_no;
    req.requirement_context = old.requirement_context;
    req.requirement_x = old.requirement_x;
    req.requirement_y = old.requirement_y;
    req.requirement_h = old.requirement_h;
    req.requirement_w = old.requirement_w;
    return req;
  }

  getNo() { return this.requirement_no; }
  setNo(no: number) { this.requirement_no = no; }
  getName() { return this.requirement_context; }
  setName(name: string) { this.requirement_context = name; }
  getShortName() { return this.requirement_context; }
  setShortName(shortName: string) { this.requirement_context = shortName; }
  getX() { return this.requirement_x; }
  setX(x: number) { this.requirement_x = x; }
  getY() { return this.requirement_y; }
  setY(y: number) { this.requirement_y = y; }
  getH() { return this.requirement_h; }
  setH(h: number) { this.requirement_h = h; }
  getW() { return this.requirement_w; }
  setW(w: number) { this.requirement_w = w; }
}

File: PF2-web1/web1-backend/src/main/java/com/example/demo/bean/Requirement.java
package com.example.demo.bean;

public class Requirement extends PdNode{
	private int requirement_no;	//需求编号
	private String requirement_context;	//需求描述
	private String requirement_shortname;
	private int requirement_x;	//位置信息
	private int requirement_y;
	private int requirement_h;
	private int requirement_w;
	public Requirement(Requirement item) {
		requirement_no = item.requirement_no;
		requirement_context = item.requirement_context;
		requirement_shortname = item.requirement_shortname;
		requirement_x = item.requirement_x;
		requirement_y = item.requirement_y;
		requirement_h = item.requirement_h;
		requirement_w = item.requirement_w;
	}

	public Requirement() {
		super();
	}

	public int getRequirement_no() {
		return requirement_no;
	}
	public void setRequirement_no(int requirement_no) {
		this.requirement_no = requirement_no;
	}
	public String getRequirement_context() {
		return requirement_context;
	}
	public void setRequirement_context(String requirement_context) {
		this.requirement_context = requirement_context;
	}
	public String getRequirement_shortname() {
		return requirement_shortname;
	}
	public void setRequirement_shortname(String requirement_shortname) {
		this.requirement_shortname = requirement_shortname;
	}
	public int getRequirement_x() {
		return requirement_x;
	}
	public void setRequirement_x(int requirement_x) {
		this.requirement_x = requirement_x;
	}
	public int getRequirement_y() {
		return requirement_y;
	}
	public void setRequirement_y(int requirement_y) {
		this.requirement_y = requirement_y;
	}
	public int getRequirement_h() {
		return requirement_h;
	}
	public void setRequirement_h(int requirement_h) {
		this.requirement_h = requirement_h;
	}
	public int getRequirement_w() {
		return requirement_w;
	}
	public void setRequirement_w(int requirement_w) {
		this.requirement_w = requirement_w;
	}
	public int getNo() {
		return requirement_no;
	}

	public void setNo(int no) {
		this.requirement_no = no;
	}

	public String getName() {
		return requirement_context;
	}

	public void setName(String name) {
		this.requirement_context = name;
	}

	public String getShortname() {
		return requirement_shortname;
	}

	public void setShortname(String shortname) {
		this.requirement_shortname = shortname;
	}

	public int getX() {
		return requirement_x;
	}

	public void setX(int x) {
		this.requirement_x = x;
	}

	public int getY() {
		return requirement_y;
	}

	public void setY(int y) {
		this.requirement_y = y;
	}

	public int getH() {
		return requirement_h;
	}

	public void setH(int h) {
		this.requirement_h = h;
	}

	public int getW() {
		return requirement_w;
	}

	public void setW(int w) {
		this.requirement_w = w;
	}
}
